### PR TITLE
DOC: Add new backends and Geospatial methods to documentation

### PR DESCRIPTION
**Pull request by xmnlab**
In this PR:

- Added initial documentation for OmniSciDB, MySQL, PySpark and SparkSQL backends
- Added initial documentation for Geo Spatial methods 
- Added link to Ibis wiki page 
- Fixed PySpark and Spark connect docstring: `**kwargs` needed to be escaped.
xref  partially #2025

### CODE REVIEW

File: docs/source/api.rst [hunk 1]
```diff
@@ -606,3 +704,74 @@ Decimal methods
 
    DecimalValue.precision
    DecimalValue.scale
+
+.. _api.geospatial:
+
+Geo Spatial methods
```
Comment (ian-r-rose) [inline docs/source/api.rst]: This is usually given as a single word "Geospatial"

Comment (xmnlab) [reply docs/source/api.rst]: thanks @ian-r-rose I will fix that!

**Comment by xmnlab** [comment]
it is done for review! :)

**Comment by xmnlab** [comment]
@scottcode @jreback could you please review this PR? thanks!

File: docs/source/backends.rst [hunk 2]
```diff
@@ -27,20 +27,13 @@ Generally speaking these backends also need to handle their own execution.
...
-TODO
-
-BigQuery
-********
-
-TODO
+- Impala
+- Clickhouse
+- BigQuery
+- HDFS
```
Comment (jreback) [inline docs/source/backends.rst]: can you add links to the actual docs for the backends

File: docs/source/backends.rst [hunk 3]
```diff
@@ -56,15 +49,9 @@ an expression. These backends tend to execute their expressions directly
...
-******
-
-TODO
-
-PostgreSQL
-**********
-
-TODO
+- PostgreSQL
+- SQLite
```
Comment (jreback) [inline docs/source/backends.rst]: same

File: docs/source/backends.rst [hunk 4]
```diff
@@ -75,7 +62,4 @@ The only existing backend that directly executes ibis expressions is the pandas
 backend. A full description of the implementation can be found in the module
 docstring of the pandas backend located in ``ibis/pandas/execution/core.py``.
 
-Pandas
-******
-
-TODO
+- Pandas
```
Comment (jreback) [inline docs/source/backends.rst]: same

File: docs/source/backends.rst (hunk 2 above)
Comment (xmnlab) [reply docs/source/backends.rst]: sounds good! I am doing that. thanks!

**Comment by xmnlab** [comment]
@jreback thanks for the review. I already applied the suggestion you made. thanks!

**Comment by xmnlab** [comment]
@jreback any more thought about this PR? let me know and if it is OK .. I will rebase to fix the conflict on release.rst

**Comment by xmnlab** [comment]
@jreback any more thought about this PR? let me know and if it is OK .. I will rebase to fix the conflict on release.rst

**Comment by jreback** [comment]
looks good @xmnlab yeah go ahead and rebase. will look at the built link then.

**Comment by xmnlab** [comment]
rebased and CI status green :)
thanks @jreback

**Comment by xmnlab** [comment]
@jreback a gentle reminder about this PR :)

**Comment by jreback** [comment]
IIRC there was a way to look at the build docs, @xmnlab can you see if there are any docs around about this?

**Comment by xmnlab** [comment]
@jreback ok I will check that today!

**Comment by xmnlab** [comment]
As I could understand ... it currently uses github pages to serve the documentation and files are pushed to that repo just after merging (as you can see here, the publish step is skipped https://dev.azure.com/ibis-project/ibis/_build/results?buildId=1749&view=logs&jobId=6a991cd6-976c-5315-84e1-201ee118e969&j=8f09edc2-e3b7-52de-126a-0225c4f3efa1&t=821d7798-77bc-5c2b-d93a-439a2b545f54). 

Currently, if you want to check the built docs you need to build it locally

let me know if I am missing anything ..

**Comment by xmnlab** [comment]
hey @jreback ! is there anything else I can do for this PR? or could it be merged? thanks!

**Comment by xmnlab** [comment]
@jreback it seems docs is working properly. thanks!

**Pull request merged**